opencog/atoms/base/Atom.h

```diff
@@ -190,6 +195,16 @@ class Atom
         return classserver().isA(at, t);
     }
 
+    /** Returns the outgoing set.
+     *
+     * @return The outgoing set.
+     */
+    virtual inline const HandleSeq& getOutgoingSet() const
+    {
+        static HandleSeq no_atoms;
+        return no_atoms;
+    }
```

Comment:
The C++ cast is free, it's the checking of the result for NULL before calling a function on that result that is not free. At any rate, a branch on zero is not expensive. It's more ugly and verbose than slow.

As to whether or not it's an error, that seems like something that can be decided rather than a given. I can certainly see myself writing code that will print out the name of all the atoms during debugging etc. And I'd not find it confusing that names didn't exist for non-nodes.

Nevertheless, I'm in no position to argue with you about this as you've got much more experience with the domain. Throwing an exception in C++ is fine if that's what you'd prefer. I'll do it.

However, this may break Python code. The code for the name property wasn't throwing exceptions before since atomspace.get_name() didn't throw, so there was no need to do a type check on the handle before calling atomspace.get_name(). If there are any problems, I'll fix the tests.